src/LibraryViewExtension/LibraryViewCustomization.cs

```diff
@@ -127,5 +133,59 @@ public Stream ToJSONStream()
         {
             return root.ToJSONStream();
         }
+
+        /// <summary>
+        /// Notifies when a resource stream is registered
+        /// </summary>
+        public event Action<string, Stream> ResourceStreamRegistered;
+
+        /// <summary>
+        /// Registers a given resource stream for a given url. If registered 
+        /// successfully the requested resource is returned from the given stream.
+        /// </summary>
+        /// <param name="urlpath">relative path of url including 
+        /// extension of the resource, e.g. /myicons/xicon.png</param>
+        /// <param name="resource">resource data stream</param>
+        /// <returns>True if the operation was successful</returns>
+        public bool RegisterResourceStream(string urlpath, Stream resource)
+        {
+            var handler = ResourceStreamRegistered;
+            if(handler != null)
+            {
+                handler(urlpath, resource);
+            }
```

Comment:
Shouldn't we be notifying *after* the stream is successfully registered? For one, on failure we don't cause the listener to incorrectly act on it; for another, `resources` would not have been updated if listeners try to get it as soon as they receive notification.